dev/notification.html

```diff
@@ -6,25 +6,159 @@
+      }
+
+      const positionSelect = document.querySelector('#position');
+      positionSelect.items = [
+        { label: 'All', value: 'all' },
+        { label: 'Random', value: 'random' },
+        { component: 'hr' },
+        { label: 'Top Stretch', value: 'top-stretch' },
+        { label: 'Top Start', value: 'top-start' },
+        { label: 'Top Center', value: 'top-center' },
+        { label: 'Top End', value: 'top-end' },
+        { label: 'Middle', value: 'middle' },
+        { label: 'Bottom Start', value: 'bottom-start' },
+        { label: 'Bottom Center', value: 'bottom-center' },
+        { label: 'Bottom End', value: 'bottom-end' },
+        { label: 'Bottom Stretch', value: 'bottom-stretch' },
+      ];
+      positionSelect.value = 'random';
+
+      const durationInput = document.querySelector('#duration');
+
+      function showNotification(position) {
+        const duration =
+          durationInput.value !== '' ? parseInt(durationInput.value) * 1000 : 1000 + Math.random() * 6000;
+        const notification = Notification.show(html`${generateRandomSentence()} <vaadin-button>Close</vaadin-button>`, {
+          position,
+          duration,
+        });
+        notification._card.querySelector('vaadin-button').addEventListener('click', () => {
+          notification.close();
+        });
+      }
+
+      document.querySelector('#show-notification').addEventListener('click', () => {
+        let position = positionSelect.value;
+
+        const posOptions = positionSelect.items.slice(3);
+        if (position === 'all') {
+          posOptions.forEach(({ value }) => {
+            showNotification(value);
+          });
+        } else {
+          if (position === 'random') {
+            position = posOptions[parseInt(Math.random() * posOptions.length)].value;
+          }
+          showNotification(position);
+        }
+      });
+    </script>
+
+    <style>
+      section {
+        padding-top: 30vh;
+        gap: 1em;
+        /* TODO remove once baseline alignment works */
+        align-items: end;
+        justify-content: center;
+      }
+
+      /* TODO should be a built-in variant */
+      #duration input {
+        text-align: end;
+      }
+
+      #duration {
+        width: 7em;
+      }
+
+      vaadin-notification-card::part(content) {
+        display: grid;
+        grid-template-columns: 1fr minmax(0, auto);
+        align-items: center;
+        gap: 1em;
+      }
+
+      vaadin-notification-card > * {
+        grid-column: 1;
+      }
+
+      vaadin-notification-card > vaadin-button {
+        grid-column: 2;
+      }
+
+      /* TODO (SAFARI-WORKAROUND-PART-2): without this, rendering isn't updated properly for the card that changes from order 1 -> 2 (max-height doesn't update) */
+      /* Either this needs to be implemented with SlotStylesMixin or we need to find a way to trigger the same behavior with shadow DOM styles */
+      vaadin-notification-card[style] {
+      }
```

Comment:
If we can't find a better solution, then this needs to be implemented as slotted styles.

Reply:
Postponed until later, no animations for now.